libr/arch/p/sbpf_cs/plugin.c

```diff
@@ -95,22 +185,394 @@ static bool decode(RArchSession *a, RAnalOp *op, RArchDecodeMask mask) {
+			if (mask & R_ARCH_OP_MASK_DISASM) {
+				op->mnemonic = r_str_newf ("uhmul64 r%d, r%d", dst_reg, src_reg);
+			}
+			return true;
+		} else if (len >= 8 && buf[0] == SBPF_INS_LMUL64_IMM && sbpf_version >= SBPF_V2) {
+			// Opcode 0x96 = LMUL64_IMM (lower half of 64×64 multiply with immediate) - v2+
+			ut8 dst_reg = buf[1] & 0x0F;
+			st32 imm = r_read_le32 (buf + 4);
+
+			op->type = R_ANAL_OP_TYPE_MUL;
+			if (mask & R_ARCH_OP_MASK_DISASM) {
+				op->mnemonic = r_str_newf ("lmul64 r%d, 0x%x", dst_reg, imm);
+			}
+			return true;
+		} else if (len >= 8 && buf[0] == SBPF_INS_LMUL64_REG && sbpf_version >= SBPF_V2) {
+			// Opcode 0x9e = LMUL64_REG (lower half of signed 64×64 multiply) - v2+
+			ut8 dst_reg = buf[1] & 0x0F;
+			ut8 src_reg = (buf[1] >> 4) & 0x0F;
+
+			op->type = R_ANAL_OP_TYPE_MUL;
+			if (mask & R_ARCH_OP_MASK_DISASM) {
+				op->mnemonic = r_str_newf ("lmul64 r%d, r%d", dst_reg, src_reg);
+			}
+			return true;
+		} else if (len >= 8 && buf[0] == SBPF_INS_HOR64 && sbpf_version >= SBPF_V2) {
+			// Opcode 0xf7 = HOR64 (horizontal or) - v2+ only
+			ut8 dst_reg = buf[1] & 0x0F;
+			st32 imm = r_read_le32 (buf + 4);
+
+			op->type = R_ANAL_OP_TYPE_OR;
+			if (mask & R_ARCH_OP_MASK_DISASM) {
+				op->mnemonic = r_str_newf ("hor64 r%d, 0x%x", dst_reg, imm);
+			}
+		} else if (len >= 8 && buf[0] == SBPF_INS_ADD64_IMM && sbpf_version >= SBPF_V1) {
+			// Opcode 0x07 = ADD64_IMM (usually used with r10 for dynamic stack frames in v1+)
+			ut8 dst_reg = buf[1] & 0x0F;
+			st32 imm = r_read_le32 (buf + 4);
+
+			if (sbpf_version >= SBPF_V1 && (imm % 64) == 0) {
+				// Valid stack frame allocation in v1+
+				op->type = R_ANAL_OP_TYPE_ADD;
+				if (mask & R_ARCH_OP_MASK_DISASM) {
+					op->mnemonic = r_str_newf ("add64 r%d, 0x%x", dst_reg, imm);
+				}
+			} else {
+				op->type = R_ANAL_OP_TYPE_ILL;
+				if (mask & R_ARCH_OP_MASK_DISASM) {
+					op->mnemonic = strdup ("invalid");
+				}
+			}
+		}  else { 
+			op->type = R_ANAL_OP_TYPE_ILL;
+			if (mask & R_ARCH_OP_MASK_DISASM) {
+				op->mnemonic = strdup ("invalid");
+			}
 		}
 	} else {
 		if (mask & R_ARCH_OP_MASK_DISASM) {
+			if (insn->id == BPF_INS_CALLX) {
+				op->type = R_ANAL_OP_TYPE_UCALL;
+
+				if (sbpf_version >= SBPF_V2) {
+					// v2+: src field (SIMD-0174)
+					ut8 src_reg = (buf[1] >> 4) & 0x0F;
+					if (mask & R_ARCH_OP_MASK_DISASM) {
+						op->mnemonic = r_str_newf ("callx r%d", src_reg);
+					}
+				} else {
+					// v0/v1: imm field
+					st32 reg_num = IMM (0);
+					if (mask & R_ARCH_OP_MASK_DISASM) {
+						op->mnemonic = r_str_newf ("callx r%d", reg_num);
+					}
+				}
+			} else if (insn->id ==  BPF_INS_EXIT ) {
+				if (sbpf_version >= SBPF_V3) {
+					op->type = R_ANAL_OP_TYPE_CALL;
+				
+					st32 imm = r_read_le32 (buf + 4);
+					const char *syscall_name = get_syscall_name (imm);
+					if (syscall_name) {
+						op->mnemonic = r_str_newf ("syscall %s", syscall_name);
+					} else {
+						op->mnemonic = r_str_newf ("syscall (invalid)");
```

Comment:
the get_syscall_name can return "(invalid)" so it will never be NULL, and the ifelse can be removed